### ISSUE TITLE
Add decorators to potentially skip tests that require optional dependencies

### ISSUE THREAD

**Issue by spencerkclark** [spencerahill/aospy#167]
Currently we have tests which require:
- multiprocess

And if we add tests of the notebook they will also require:
- matplotlib
- runipy or pytest-ipynb

None of these are required dependencies for aospy, so we should only run those tests if those libraries are present on someone's system.  xarray has some good examples for how to do this in: [tests/__init__.py](https://github.com/pydata/xarray/blob/master/xarray/tests/__init__.py).

**Comment by spencerahill** [comment]
See https://github.com/spencerahill/aospy/pull/166#issuecomment-290457309

@spencerkclark if you think I'm on the right track there, then I can add the same logic for multiprocess and matplotlib

**Comment by spencerkclark** [comment]
It would be great if you could take care of multiprocess (I took care of matplotlib already; the notebook test is the only one that requires it).

Simultaneously could you add a new CI environment, called like `environment-py27-min.yml` (e.g. [xarray's](https://github.com/pydata/xarray/blob/master/ci/requirements-py27-min.yml)) that includes only our minimum requirements?  This will ensure that our import skips are operating as desired.

**Comment by spencerahill** [comment]
Good call, will do.

**Comment by spencerahill** [comment]
C.f. #168, need to catch pytest-catchlog also for `test_automate`

**Comment by spencerahill** [comment]
After #172, multiprocess is no longer needed.
